### PR TITLE
[WFLY-15598] WebServices UsernameToken profile integration with Elytron

### PR DESCRIPTION
**Pull request by skyllarr**
https://issues.redhat.com/browse/WFLY-15598
Relates to : https://issues.redhat.com/browse/JBWS-4251

Requires:
 jbossws/jbossws-spi#11
 jbossws/jbossws-cxf#179

### CODE REVIEW

**Comment by darranl** [comment]
Hi @Skyllarr it looks like this has a merge conflct.

Although this is developed under WFLY-15598 I don't think this is sufficient on it;s own to resolve the Jira?  i.e. we need the other part of the changes coming via the WS module?

**Comment by Skyllarr** [comment]
@darranl Rebased. Yes, this issue is not sufficient to resolve the Jira, we need the jbossws-cxf changes[1] and jbossws-spi[2] changes as well https://issues.redhat.com/browse/JBWS-4251 .

[1] https://github.com/jbossws/jbossws-cxf/pull/179
[2] https://github.com/jbossws/jbossws-spi/pull/11

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-server/main/module.xml [hunk 1]
```diff
@@ -69,6 +70,8 @@
         </module>
         <module name="org.jboss.logging" />
         <module name="org.apache.ws.security" />
+        <module name="org.picketbox" />
```
Comment (bstansberry) [inline ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-server/main/module.xml:73]: Is this added back by mistake?

Comment (Skyllarr) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-server/main/module.xml]: @bstansberry Yes that was a mistake, fixed now. Thank you

**Comment by jamezp** [comment]
I placed a hold on this because of the requirement for a release of the jbossws-api with https://github.com/jbossws/jbossws-spi/pull/11. The other option is to remove the `@Override` annotation from the method.

**Comment by darranl** [comment]
Does the interface in WS have a default implementation of the method?  Otherwise the component upgrade will have the opposite problem of the method not being implemented - removing the override annotation at least temporarily may be easiest to get the first part in.

**Comment by Skyllarr** [comment]
@jamezp @darranl I added the [default impl PR](https://github.com/jbossws/jbossws-spi/pull/12/files) to WS and removed the override in this PR

**Comment by jamezp** [comment]
It looks like we still have a chicken and egg problem.
```
[ERROR] Failed to execute goal org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:5.2.10.Final:build-feature-pack (feature-pack-build) on project wildfly-ee-galleon-pack: Package org.jboss.ws.cxf.jbossws-cxf-server has unsatisifed external dependency on package org.picketbox -> [Help 1]
[ERROR] Failed to execute goal org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:5.2.10.Final:build-feature-pack (feature-pack-build) on project wildfly-preview-feature-pack: Package org.jboss.ws.cxf.jbossws-cxf-server has unsatisifed external dependency on package org.picketbox -> [Help 1]
```

**Comment by Skyllarr** [comment]
> ~It looks like we still have a chicken and egg problem.~
> 
> ```
> [ERROR] Failed to execute goal org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:5.2.10.Final:build-feature-pack (feature-pack-build) on project wildfly-ee-galleon-pack: Package org.jboss.ws.cxf.jbossws-cxf-server has unsatisifed external dependency on package org.picketbox -> [Help 1]
> [ERROR] Failed to execute goal org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:5.2.10.Final:build-feature-pack (feature-pack-build) on project wildfly-preview-feature-pack: Package org.jboss.ws.cxf.jbossws-cxf-server has unsatisifed external dependency on package org.picketbox -> [Help 1]
> ```
> 
> It looks like the `org.picketbox` dependency needs to be removed as well. 

@jamezp ah, I somehow added it by mistake again. I removed it now.

**Review by jamezp — APPROVED** [review]
This looks good to me and is required for moving forward with the CXF Jakarta EE 9+ updates.

**Comment by jimma** [comment]
@Skyllarr  I already released jbossws-spi 3.4.0.Final. You can include the jbossws-spi upgrade in this PR.

**Comment by Skyllarr** [comment]
> @Skyllarr I already released jbossws-spi 3.4.0.Final. You can include the jbossws-spi upgrade in this PR.

@jimma ok,I included the version upgrade and added back Override annotation.

**Comment by jamezp** [comment]
The same tests keep failing. They seem related to this PR.

**Comment by Skyllarr** [comment]
@jamezp ok thank you. So maybe we should split the jbossws-spi upgrade from this PR to see if that is the problem. @jimma WDYT?

**Comment by jimma** [comment]
@Skyllarr These are jms test failures. It looks it's not relevant to this PR.

**Comment by Skyllarr** [comment]
> @Skyllarr These are jms test failures. It looks it's not relevant to this PR.

@jimma  Yeah but they were not failing before I added the jbossws-spi upgrade. I can try retest again and check if other PRs have them failing

**Comment by bstansberry** [comment]
@Skyllarr @jamezp @jimma I just did a retest as the last CI of this was a couple weeks old. If it passes or only has unrelated failures, please ping the mergers in https://wildfly.zulipchat.com/#narrow/stream/274477-Pull-requests so we can get it merged. The PR queue is long so a reminder ping would be good.

**Comment by bstansberry** [comment]
Thanks, @Skyllarr

**Pull request merged**